Updated documentation for OshiSysMonitor

hardikbajaj opened the pull request:
<!-- Thanks for trying to help us make Apache Druid be the best it can be! Please fill out as much of the following information as is possible (where relevant, and remove it when irrelevant) to help make the intention and scope of this PR clear in order to ease review. -->

<!-- Please read the doc for contribution (https://github.com/apache/druid/blob/master/CONTRIBUTING.md) before making this PR. Also, once you open a PR, please _avoid using force pushes and rebasing_ since these make it difficult for reviewers to see what you've changed in response to their reviews. See [the 'If your pull request shows conflicts with master' section](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#if-your-pull-request-shows-conflicts-with-master) for more details. -->

Fixes #14359

<!-- Replace XXXX with the id of the issue fixed in this PR. Remove this section if there is no corresponding issue. Don't reference the issue in the title of this pull-request. -->

<!-- If you are a committer, follow the PR action item checklist for committers:
https://github.com/apache/druid/blob/master/dev/committer-instructions.md#pr-and-issue-action-item-checklist-for-committers. -->

### Description
Updates documentation for OshiSysMontior
<!-- Describe the goal of this PR, what problem are you fixing. If there is a corresponding issue (referenced above), it's not necessary to repeat the description here, however, you may choose to keep one summary sentence. -->

<!-- Describe your patch: what did you change in code? How did you fix the problem? -->

<!-- If there are several relatively logically separate changes in this PR, create a mini-section for each of them. For example: -->

#### Fixed the bug ...
#### Renamed the class ...
#### Added a forbidden-apis entry ...

<!--
In each section, please describe design decisions made, including:
 - Choice of algorithms
 - Behavioral aspects. What configuration values are acceptable? How are corner cases and error conditions handled, such as when there are insufficient resources?
 - Class organization and design (how the logic is split between classes, inheritance, composition, design patterns)
 - Method organization and design (how the logic is split between methods, parameters and return types)
 - Naming (class, method, API, configuration, HTTP endpoint, names of emitted metrics)
-->


<!-- It's good to describe an alternative design (or mention an alternative name) for every design (or naming) decision point and compare the alternatives with the designs that you've implemented (or the names you've chosen) to highlight the advantages of the chosen designs and names. -->

<!-- If there was a discussion of the design of the feature implemented in this PR elsewhere (e. g. a "Proposal" issue, any other issue, or a thread in the development mailing list), link to that discussion from this PR description and explain what have changed in your final design compared to your original proposal or the consensus version in the end of the discussion. If something hasn't changed since the original discussion, you can omit a detailed discussion of those aspects of the design here, perhaps apart from brief mentioning for the sake of readability of this PR description. -->

<!-- Some of the aspects mentioned above may be omitted for simple and small changes. -->

#### Release note
<!-- Give your best effort to summarize your changes in a couple of sentences aimed toward Druid users. 

If your change doesn't have end user impact, you can skip this section.

For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).

-->


<hr>

##### Key changed/added classes in this PR


<hr>

<!-- Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items from the checklist below are strictly necessary, but it would be very helpful if you at least self-review the PR. -->

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

hardikbajaj commented:
I didn't noticed wrong formatter formatted this file on my local :( . I'll get back with updated documentation

hardikbajaj commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
-The `taskIngestionMode` dimension includes the following modes: 
-* `APPEND`: a native ingestion job appending to existing segments 
+The `taskIngestionMode` dimension includes the following modes:
+
```
This blank line is coming again with auto formatting. Do you want me to remove this ?

ektravel commented on docs/operations/metrics.md:
```diff
@@ -23,7 +23,8 @@ title: "Metrics"
 
 
-You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
+You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring
```
```suggestion
You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -23,7 +23,8 @@ title: "Metrics"
 
-You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
+You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring
+query execution, ingestion, coordination, and so on.
```
Remove line 27.

ektravel commented on docs/operations/metrics.md:
```diff
@@ -40,7 +41,8 @@ Metrics may have additional dimensions beyond those listed above.
 ### Router
-|Metric|Description|Dimensions|Normal value|
+
+|Metric|Description|Dimensions|Normal Value|
```
```suggestion
|Metric|Description|Dimensions|Normal value|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -135,7 +137,8 @@ Metrics may have additional dimensions beyond those listed above.
 #### Memcached only metrics
 
-Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to from the cache retrieval layer.
+Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to
```
```suggestion
Memcached client metrics are reported as per the following. These metrics come directly from the client.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -135,7 +137,8 @@ Metrics may have additional dimensions beyond those listed above.
 
-Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to from the cache retrieval layer.
+Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to
+from the cache retrieval layer.
```
Remove line 141.

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
-|`true`|`true `|Invalid combination, exception thrown.|
-|`false`|`false`|`REPLACE_LEGACY`. The default for JSON-based batch ingestion. |
-|`false`|`true`|`REPLACE`|
+| `isAppendToExisting` | `isDropExisting` | mode |
```
```suggestion
|`isAppendToExisting`|`isDropExisting`|mode|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
-|`false`|`false`|`REPLACE_LEGACY`. The default for JSON-based batch ingestion. |
-|`false`|`true`|`REPLACE`|
+| `isAppendToExisting` | `isDropExisting` | mode |
+|----------------------|-------------------|------|
```
```suggestion
|--------------------|----------------|----|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
-|`false`|`true`|`REPLACE`|
+| `isAppendToExisting` | `isDropExisting` | mode |
+|----------------------|-------------------|------|
+| `true`               | `false` | `APPEND`|
```
```suggestion
|`true`|`false`|`APPEND`|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
+| `isAppendToExisting` | `isDropExisting` | mode |
+|----------------------|-------------------|------|
+| `true`               | `false` | `APPEND`|
+| `true`               | `true  ` | Invalid combination, exception thrown. |
```
```suggestion
|`true`|`true`|Invalid combination, exception thrown.|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
+|----------------------|-------------------|------|
+| `true`               | `false` | `APPEND`|
+| `true`               | `true  ` | Invalid combination, exception thrown. |
+| `false`              | `false` | `REPLACE_LEGACY` (this is the default for native batch ingestion). |
```
```suggestion
|`false`|`false`|`REPLACE_LEGACY` (This is the default for native batch ingestion.)|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -162,24 +165,25 @@ If SQL is enabled, the Broker will emit the following metrics for SQL.
+| `true`               | `false` | `APPEND`|
+| `true`               | `true  ` | Invalid combination, exception thrown. |
+| `false`              | `false` | `REPLACE_LEGACY` (this is the default for native batch ingestion). |
+| `false`              | `true` | `REPLACE`|
```
```suggestion
|`false`|`true`|`REPLACE`|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -239,9 +243,10 @@ batch ingestion emit the following metrics. These metrics are deltas for each em
-|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
-|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
-If the JVM does not support CPU time measurement for the current thread, `ingest/merge/cpu` and `ingest/persists/cpu` will be 0.
+|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on coordinator cycle time.|
```
```suggestion
|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
 ## Shuffle metrics (Native parallel task)
 
-The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`
+The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor`
```
```suggestion
The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
 
-The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`
+The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor`
+in `druid.monitoring.monitors`
```
Remove line 288.

ektravel replied:
```suggestion
in `druid.monitoring.monitors`.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -279,7 +284,8 @@ If the JVM does not support CPU time measurement for the current thread, `ingest
-The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`
+The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor`
+in `druid.monitoring.monitors`
 See [Enabling Metrics](../configuration/index.md#enabling-metrics) for more details.
```
```suggestion
See [Enabling metrics](../configuration/index.md#enabling-metrics) for more details.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
-## Sys
+## Sys [Deprecated]
+
+> SysMonitor will be deprecated due to its reliance on the sigar library, which is no longer maintained.
```
```suggestion
> SysMonitor will be deprecated due to its reliance on the Sigar library, which is no longer maintained.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
+## Sys [Deprecated]
+
+> SysMonitor will be deprecated due to its reliance on the sigar library, which is no longer maintained.
+> A new Monitor called [OshiSysMonitor](#oshisysmonitor), which provides similar functionality but is based on the oshi
```
```suggestion
> Instead, use the new OSHI monitor called [OshiSysMonitor](#oshisysmonitor). The new monitor has a wider support for different machine architectures including ARM instances.
```

ektravel commented on docs/operations/metrics.md:
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
+
+> SysMonitor will be deprecated due to its reliance on the sigar library, which is no longer maintained.
+> A new Monitor called [OshiSysMonitor](#oshisysmonitor), which provides similar functionality but is based on the oshi
+> library. We encourage all users to switch to OshiSysMonitor as soon as possible.
```
Remove line 415.

ektravel commented on docs/operations/metrics.md:
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
-|`sys/mem/max`|Memory max||Varies|
-|`sys/storage/used`|Disk space used|`fsDirName`|Varies|
-|`sys/cpu`|CPU used|`cpuName`, `cpuTime`|Varies|
+| Metric                 | Description                                                                          |Dimensions|Normal Value|
```
If `SysMonitor` is deprecated, should these metrics be removed?

Also, please fix the table formatting. Do not add whitespace to make columns longer. Trim excessive white space between columns.

ektravel commented on docs/operations/metrics.md:
```diff
@@ -402,32 +408,82 @@ The following metric is only available if the `EventReceiverFirehoseMonitor` mod
+
+These metrics are only available if the `OshiSysMonitor` module is included.
+
+| Metric                     | Description                                                                          | Dimensions                           | Normal Value        |
```
Please fix the table formatting.

Do not add whitespace to make columns longer. Trim excessive white space between columns.

ektravel commented on website/.spelling:
```diff
@@ -2033,6 +2033,7 @@ org.apache.druid
 org.apache.druid.jetty.RequestLog
 org.apache.hadoop
 overlord.html
+oshi
```
```suggestion
OSHI
```

ektravel commented on website/.spelling:
```diff
@@ -2051,6 +2052,7 @@ slf4j
 sql
 sqlQuery
 successfulSending
+sigar
```
```suggestion
[S]igar
```

ektravel reviewed (commented):
Reviewed documentation portion of the PR.

hardikbajaj commented:
@ektravel @abhishekagarwal87 , This PR got closed accidentally, I have moved it to a new PR now https://github.com/apache/druid/pull/14912 and addressed all the comments there.
